packages/amazon-ssm-agent/amazon-ssm-agent.service

```diff
@@ -0,0 +1,18 @@
+[Unit]
+Description=amazon-ssm-agent
+
+[Service]
+Type=simple
+ExecStart=/usr/bin/amazon-ssm-agent
+KillMode=process
+
+# Restart the agent regardless of whether it crashes (and returns a non-zero result code) or if
+# is terminated normally (e.g. via 'kill -HUP').  Delay restart so that the agent is less likely
+# to restart during a reboot initiated by a script. If the agent exits with status 194 (reboot
+# requested), don't restart at all.
+Restart=always
+RestartPreventExitStatus=194
+RestartSec=90
```

Comment:
This logic doesn't make sense to me. A properly initiated reboot should cause systemd to transition to `shutdown.target` which will cause this service to stop (because of default dependencies).

I'm OK with the special treatment for the `194` exit code, but the delay isn't good. It means if the agent crashes for whatever reason it will take 90 seconds to come back up, which is an eternity if SSM is the only mechanism for managing the node.

Reply:
I'm fine with lowering the restart time, but we would be altering the default SSM agent configuration. Are we okay with this? If so, what restart time do you recommend?

Reply:
> I'm fine with lowering the restart time, but we would be altering the default SSM agent configuration. Are we okay with this? If so, what restart time do you recommend?

I don't have a specific recommendation, just the feedback that 90 seconds is too long. You can look at other `RestartSec` values in `packages/*/*.service` to get a sense of the acceptable ranges.

Whatever value you pick, you should back up with data along the lines of "I inspected the ssm agent source code and I'm confident this is fine because ..." or "I tested the script-initiated reboot path in Bottlerocket, inspected the system journal, and confirmed that no unexpected restarts were triggered even with a lower restart time."